src/blueprint/azext_blueprint/_params.py

```diff
@@ -0,0 +1,215 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+# pylint: disable=line-too-long
+# pylint: disable=too-many-lines
+# pylint: disable=too-many-statements
+
+from azure.cli.core.commands.parameters import (
+    tags_type,
+    get_enum_type,
+    get_location_type
+)
+from knack.arguments import CLIArgumentType
+from argcomplete.completers import FilesCompleter
+from ._validators import scope_validator
+
+parameter_type = CLIArgumentType(
+    options_list=['--parameters'],
+    help='Parameters in JSON format or file path to Json file with "@" prefix.'
+)
+
+template_type = CLIArgumentType(
+    options_list=['--template'],
+    help='ARM template in JSON format or file path to Json file with "@" prefix.'
+)
+
+
+def load_arguments(self, _):
+
+    with self.argument_context('blueprint', validator=scope_validator) as c:
+        c.argument('management_group', help='The management group for the scope of the resource. The scope could either be a subscription or a management group. For blueprint assignments management group scope is reserved for future use. If management-group is not specified, will use subscription as scope')
+        c.ignore('scope')
+
+    with self.argument_context('blueprint create') as c:
+        c.argument('blueprint_name', options_list=['--name', '-n'], help='Name of the blueprint definition.')
+        c.argument('display_name', help='One-liner string explain this resource.')
+        c.argument('description', help='Multi-line explain this resource.')
+        c.argument('target_scope', arg_type=get_enum_type(['subscription', 'managementGroup']), help='The scope where this blueprint definition can be assigned.')
+        c.argument('parameters', arg_type=parameter_type, help='Parameters required by this blueprint definition.')
+
+    with self.argument_context('blueprint import') as c:
+        c.argument('blueprint_name', options_list=['--name', '-n'], help='Name of the blueprint definition.')
+        c.argument('input_path', help='The directory path for json definitions of blueprint and artifacts. Artifacts json files should be in input-path/artifacts/.', completer=FilesCompleter())

```

Comment:
When testing this command, it is not clear that input_path should NOT be an absolute path. For example, while running in my python virtual environment, attempting to give an absolute path to a json file to the import command fails:  az blueprint import --name test --input-path " C:\Users\stevens\Projects\azure-cli-extensions\src\blueprint\azext_blueprint\tests\latest\input\import_with_arm"
OSError: [Errno 22] Invalid argument: 'C:\\Users\\stevens\\Projects\\env\\ C:\\Users\\stevens\\Projects\\azure-cli-extensions\\src\\blueprint\\azext_blueprint\\tests\\latest\\input\\import_with_arm\\blueprint.json'

It will work if I give it the correct path relative to the environment of course: az blueprint import --name test --input-path "..\azure-cli-extensions\src\blueprint\azext_blueprint\tests\latest\input\import_with_arm"

Reply:
Which shell did you use? I cannot reproduce with cmd or powershell for the absolute path issue in a virtual env.

Reply:
I'm using PS 6.2.4. I see what a happened here, though. If there is any leading white space in the input path, you get the error were it attempts to treat the input string as a relative path, which I assume is the expected behavior then.

Reply:
good to know about this.